### PR TITLE
render children inside AppBar

### PR DESCRIPTION
**Pull request by astiag**
this PR will render children inside AppBar component

### CODE REVIEW

**Comment by fzaninotto** [comment]
please add documentation and example usage

**Comment by alanpoulain** [comment]
Related to this: https://github.com/marmelab/react-admin/issues/2534

**Comment by fzaninotto** [comment]
Let me rephrase my remark: please add documentation and example usage in the files of your PR (i.e. modify the react-admin documentation, not only the code).

**Comment by alanpoulain** [comment]
I can but I am not the author of this PR :sweat_smile:

**Comment by astiag** [comment]
I know your rules about having everything documented.
but the thing is you already have it [here](https://github.com/marmelab/react-admin/blob/master/docs/Theming.md#using-a-custom-appbar)

btw your docs regarding to adding children to `AppBar` will cause some problems

```
const MyAppBar = props => (
    <AppBar {...props}>
        <Toolbar>
            <Typography variant="title" id="react-admin-title" />
        </Toolbar>
    </AppBar>
);
```
in this case you will have `Toolbar` inside `Toolbar`. [see here](https://github.com/marmelab/react-admin/blob/1f47c28b87b3b63b1455894a05a2d818d102c283/packages/ra-ui-materialui/src/layout/AppBar.js#L68)

so I also will fix this one in a new commit in this PR

**Comment by alanpoulain** [comment]
@astiag No I don't think it's the case. You are mixing up `AppBar` from `@material-ui` and `AppBar` from `ra-ui-materialui`. The first can take children (and a `Toolbar`), the second can't.

**Comment by astiag** [comment]
@alanpoulain you are right. didnt notice that.
anyway I'll add something in documents about having children inside `AppBar`

**Comment by astiag** [comment]
@fzaninotto please review again

File: docs/Theming.md [hunk 1]
```diff
@@ -409,6 +409,22 @@ const MyAppBar = props => <AppBar {...props} userMenu={<MyUserMenu />} />;
 const MyLayout = props => <Layout {...props} appBar={MyAppBar} />;
 ```
 
+`AppBar` can accept children so you are able to add more custom components in it.
```
Comment (djhi) [inline docs/Theming.md]: This should probably go inside the `Using a Custom AppBar` section.

Comment (fzaninotto) [reply docs/Theming.md]: 👍

**Comment by fzaninotto** [comment]
Any progress?

File: docs/Theming.md (hunk 1 above)
Comment (astiag) [reply docs/Theming.md]: will change it asap

**Comment by BartoGabriel** [comment]
I think it would be good to combine this feature, with the possibility of disabling the LoadingIndicator or the UserMenu of the AppBar.
Since the user having the ability to customize AppBar, he could create their own components, without having to make a new AppBar.
With two properties it could be solved.

**Comment by astiag** [comment]
> Any progress?
@fzaninotto @djhi I've moved the description to `Using a Custom AppBar` section

File: docs/Theming.md (hunk 1 above)
Comment (astiag) [reply docs/Theming.md]: done

**Comment by fzaninotto** [comment]
Thanks for the PR, I've finally decided to do a slightly different implementation (replacing the content with children instead of appending the children to the default content) in #2777. So I'll close this PR.
